File: BlackJack.ViewModels/GameViews/GetAllStepOfBotsByGameIdGameView.cs
using BlackJack.ViewModels.EnumViews;
using System;
using System.Collections.Generic;

namespace BlackJack.ViewModels.GameViews
{
    public class GetAllStepOfBotsByGameIdGameView
    {
        public List<BotStepGetAllStepOfBotsByGameIdGameViewItem> BotSteps { get; set; }

        public GetAllStepOfBotsByGameIdGameView()
        {
            BotSteps = new List<BotStepGetAllStepOfBotsByGameIdGameViewItem>();
        }
    }

    public class BotStepGetAllStepOfBotsByGameIdGameViewItem
    {
        public Guid Id { get; set; }
        public SuiteTypeEnumView Suite { get; set; }
        public RankTypeEnumView Rank { get; set; }
        public BotGetAllStepOfBotsByGameIdGameView Bot { get; set; }
    }

    public class BotGetAllStepOfBotsByGameIdGameView
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public decimal Balance { get; set; }
        public decimal Bet { get; set; }
    }
}

File: BlackJack.WEB/ClientApp/src/app/shared/entities/game.views/get-all-step-of-bots-by-game-id.game.view.ts
import { SuiteType } from '../../enums/suite-type';
import { RankType } from '../../enums/rank-type';

export class GetAllStepOfBotsByGameIdGameView {
  botSteps?: Array<BotStepGetAllStepOfBotsByGameIdGameViewItem>;

  constructor() {
    this.botSteps = new Array<BotStepGetAllStepOfBotsByGameIdGameViewItem>();
  }
}
export class BotStepGetAllStepOfBotsByGameIdGameViewItem {
  id?: string;
  suite?: SuiteType;
  rank?: RankType;
  bot?: BotGetAllStepOfBotsByGameIdGameView;

  constructor() {
    this.bot = new BotGetAllStepOfBotsByGameIdGameView();
  }
}

export class BotGetAllStepOfBotsByGameIdGameView {
  id?: string;
  name?: string;
  balance?: number;
  bet?: number;
}
